presto-memory-context/src/main/java/com/facebook/presto/memory/context/AggregatedMemoryContext.java

```diff
@@ -79,19 +78,19 @@ public synchronized long getBytes()
         if (parentMemoryContext != null) {
             // update the parent before updating usedBytes as it may throw a runtime exception (e.g., ExceededMemoryLimitException)
             future = parentMemoryContext.updateBytes(bytes);
-            usedBytes += bytes;
+            usedBytes = addExact(usedBytes, bytes);
             return future;
         }
 
-        // At this point this is root
+        // at this point this is root
         if (reservationHandler != null) {
             future = reservationHandler.reserveMemory(bytes);
         }
 
-        usedBytes += bytes;
+        usedBytes = addExact(usedBytes, bytes);
```

Comment:
This addresses overflow, thanks. What about going negative?

(I know there is a check in `SimpleLocalMemoryContext` but i'd prefer `AggregatedMemoryContext` itself remained responsible for guarding its state)